tests/spec/integration_spec.lua

```
@@ -0,0 +1,454 @@
+      }
+    }
+    
+    local success = pcall(caramba.setup, custom_config)
+    assert.is_true(success, "Should setup with custom config")
+  end)
+  
+  it("should handle complete context extraction workflow", function()
+    reset_state()
+    caramba.setup()
+    
+    vim.bo[1] = { filetype = "lua" }
```

Luacheck (1.2.0)

[warning] mutating non-standard global variable 'vim'

(W112)